src/client/common/configSettings.ts

```diff
@@ -1,620 +1,704 @@
-        );
-        this.disposables.push(
-            this.workspace.onDidChangeConfiguration((event: ConfigurationChangeEvent) => {
-                if (event.affectsConfiguration('python')) {
-                    onDidChange();
-                }
-            })
-        );
-
-        const initialConfig = this.workspace.getConfiguration('python', this.workspaceRoot);
-        if (initialConfig) {
-            this.update(initialConfig);
-        }
-    }
-    @debounceSync(1)
-    protected debounceChangeNotification() {
-        this.changed.fire();
-    }
-}
-
-function getAbsolutePath(pathToCheck: string, rootDir: string): string {
-    // tslint:disable-next-line:prefer-type-cast no-unsafe-any
-    pathToCheck = untildify(pathToCheck) as string;
-    if (isTestExecution() && !pathToCheck) {
-        return rootDir;
-    }
-    if (pathToCheck.indexOf(path.sep) === -1) {
-        return pathToCheck;
-    }
-    return path.isAbsolute(pathToCheck) ? pathToCheck : path.resolve(rootDir, pathToCheck);
-}
-
-function getPythonExecutable(pythonPath: string): string {
-    // tslint:disable-next-line:prefer-type-cast no-unsafe-any
-    pythonPath = untildify(pythonPath) as string;
-
-    // If only 'python'.
-    if (
-        pythonPath === 'python' ||
-        pythonPath.indexOf(path.sep) === -1 ||
-        path.basename(pythonPath) === path.dirname(pythonPath)
-    ) {
-        return pythonPath;
-    }
-
-    if (isValidPythonPath(pythonPath)) {
-        return pythonPath;
-    }
-    // Keep python right on top, for backwards compatibility.
-    // tslint:disable-next-line:variable-name
-    const KnownPythonExecutables = ['python', 'python4', 'python3.6', 'python3.5', 'python3', 'python2.7', 'python2'];
-
-    for (let executableName of KnownPythonExecutables) {
-        // Suffix with 'python' for linux and 'osx', and 'python.exe' for 'windows'.
-        if (IS_WINDOWS) {
-            executableName = `${executableName}.exe`;
-            if (isValidPythonPath(path.join(pythonPath, executableName))) {
-                return path.join(pythonPath, executableName);
-            }
-            if (isValidPythonPath(path.join(pythonPath, 'scripts', executableName))) {
-                return path.join(pythonPath, 'scripts', executableName);
-            }
-        } else {
-            if (isValidPythonPath(path.join(pythonPath, executableName))) {
-                return path.join(pythonPath, executableName);
-            }
-            if (isValidPythonPath(path.join(pythonPath, 'bin', executableName))) {
-                return path.join(pythonPath, 'bin', executableName);
-            }
-        }
-    }
-
-    return pythonPath;
-}
-
-function isValidPythonPath(pythonPath: string): boolean {
-    try {
-        const output = child_process.execFileSync(pythonPath, ['-c', 'print(1234)'], { encoding: 'utf8' });
-        return output.startsWith('1234');
-    } catch (ex) {
-        return false;
-    }
-}
+'use strict';

```

Comment:
```suggestion
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

'use strict';
```

Is this missing the copyright at the top?

Reply:
That is OK if it is missing on a lot of files. we can just add it separately.

Reply:
I think it's missing from the beginning. We probably shouldn't add it as Don authored the file when he was not affiliated with Microsoft.